code/modules/reagents/newchem/medicine.dm

```diff
@@ -0,0 +1,711 @@
+#define SOLID 1
+#define LIQUID 2
+#define GAS 3
+
+#define REM REAGENTS_EFFECT_MULTIPLIER
+
+datum/reagent/silver_sulfadiazine
+	name = "Silver Sulfadiazine"
+	id = "silver_sulfadiazine"
+	description = "On touch, quickly heals burn damage. Basic anti-burn healing drug. On ingestion, deals minor toxin damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 2
+
+datum/reagent/silver_sulfadiazine/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume, var/show_message = 1)
+	if(iscarbon(M))
+		if(method == TOUCH)
+			M.adjustFireLoss(-volume)
+			if(show_message)
+				M << "<span class='notice'>You feel your burns healing!</span>"
+			M.emote("scream")
+		if(method == INGEST)
+			M.adjustToxLoss(0.5*volume)
+			if(show_message)
+				M << "<span class='notice'>You probably shouldn't have eaten that. Maybe you should of splashed it on, or applied a patch?</span>"
+	..()
+	return
+
+datum/reagent/silver_sulfadiazine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustFireLoss(-2*REM)
+	..()
+	return
+
+datum/reagent/styptic_powder
+	name = "Styptic Powder"
+	id = "styptic_powder"
+	description = "On touch, quickly heals brute damage. Basic anti-brute healing drug. On ingestion, deals minor toxin damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 2
+
+datum/reagent/styptic_powder/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume, var/show_message = 1)
+	if(iscarbon(M))
+		if(method == TOUCH)
+			M.adjustBruteLoss(-volume)
+			if(show_message)
+				M << "<span class='notice'>You feel your wounds knitting back together!</span>"
```

Comment:
Maybe check for damage first, before you make the mob scream when their nothing closes up.